sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -69,10 +69,41 @@ object Literal {
     case f: Float => Literal(f, FloatType)
     case b: Byte => Literal(b, ByteType)
     case s: Short => Literal(s, ShortType)
-    case s: String => Literal(UTF8String.fromString(s), StringType)
-    case s: UTF8String => Literal(s, StringType)
-    case c: Char => Literal(UTF8String.fromString(c.toString), StringType)
-    case ac: Array[Char] => Literal(UTF8String.fromString(String.valueOf(ac)), StringType)
+    case s: String =>
```

Comment:
Do you need `== CollationFactory.SPARK_INTERNAL_COLLATION_ID` check?
Shouldn't `StringType(CollationFactory.collationNameToId(SQLConf.get.defaultCollation)` work even for collationId = 0?

Reply:
It will work, but it will break backwards compatibility until we implement Casting. All rules only match object StringType for now, but not the class as well.

Reply:
I still don't get it? `StringType(0)` should match against `StringType`. I just tried this:

```
    val s = StringType(0)
    s match {
      case StringType => assert(true)
      case _ => assert(false)
    }
```